### ISSUE TITLE
[Feature] Allow setting of WorkDistributionStrategy

### ISSUE THREAD

**Issue by mrinalsharma** [jobrunr/jobrunr#176]
**Is your feature request related to a problem? Please describe.**
Currently, JobZooKeeper only allows using BasicWorkDistributionStrategy. As a developer, I would like to define my own strategy to take up the new Job.

**Describe the solution you'd like**
Allow BackgroundJobServer() constructor to take WorkDistributionStrategy as a parameter.  Set member workDistributionStrategy in JobZooKeeper with the passed value.

**Comment by rdehuyss** [comment]
This is already possible using the BackgroundJobServerConfiguration class.

**Comment by mrinalsharma** [comment]
I am sorry if I am ignorant, FixedSizeBackgroundJobServerWorkerPolicy only allows to set WorkerCount during initial BackgroundJobServer configuration. It doesn't allow to take more jobs or fewer jobs based on any other criteria during runtime. Is WorkDistributionStrategy not meant to decide Background Server capacity to take more work during runtime?

**Comment by rdehuyss** [comment]
There is also a method called `andBackgroundJobServerWorkerPolicy(BackgroundJobServerWorkerPolicy backgroundJobServerPolicy)`. (It was named differently though). There you can then implement `toWorkDistributionStrategy(BackgroundJobServer backgroundJobServer)`. 

Perhaps let me rephrase the question: what are you trying to achieve? Based on which criteria will you decide to take more work. If it's helpful for you, it may also be helpful for other users.

**Comment by mrinalsharma** [comment]
Currently, I am planning to add a relatively simple strategy like deciding work pick up based on available server Memory and current CPU load.  Some tasks can be CPU bound and some tasks can be I/O bound, so if a strategy with available Memory and current CPU load is applied, the Background server may be able to take more tasks by utilizing resources to the fullest.

**Comment by rdehuyss** [comment]
Well, I did various tests already (in the pro version I'm working on workdistributionstrategy that is determined by machine learning) and to be honest, the default is actually quite good. I'm curious in your results but I don't think you will have major improvements.

Do let me know your results as I'm interested.
